contribs/src/main/scala/io/delta/storage/DynamoDBLogStore.scala

```diff
@@ -0,0 +1,209 @@
+      client.putItem(entry.asPutItemRequest(tableName, overwrite))
+    } catch {
+      case e: ConditionalCheckFailedException =>
+        logDebug(e.toString)
+        throw new java.nio.file.FileAlreadyExistsException(
+          entry.absoluteJsonPath.toString
+        )
+    }
+  }
+
+  override protected def getDbEntry(
+      absoluteTablePath: Path,
+      absoluteJsonPath: Path
+  ): Option[LogEntry] = {
+    Option(
+      client
+        .getItem(
+          new GetItemRequest(
+            tableName,
+            Map(
+              "tablePath" -> new AttributeValue(absoluteTablePath.toString),
+              "fileName" -> new AttributeValue(absoluteJsonPath.getName)
+            ).asJava
+          )
+          .withConsistentRead(true)
+        )
+        .getItem
+    ).map(itemToDbEntry)
+  }
+
+  override protected def getLatestDbEntry(
+      tablePath: Path
+  ): Option[LogEntry] = {
+    client
+      .query(
+        new QueryRequest(tableName)
+          .withConsistentRead(true)
+          .withScanIndexForward(false)
+          .withLimit(1)
+          .withKeyConditions(
+            Map(
+              "tablePath" -> new Condition()
+                .withComparisonOperator(ComparisonOperator.EQ)
+                .withAttributeValueList(new AttributeValue(tablePath.toString))
+            ).asJava
+          )
+      )
+      .getItems
+      .iterator
+      .asScala
+      .map(itemToDbEntry)
+      .find(_ => true)
+  }
+
+  def itemToDbEntry(
+      item: java.util.Map[String, AttributeValue]
+  ): LogEntry = LogEntry(
+    tablePath = new Path(item.get("tablePath").getS),
+    fileName = item.get("fileName").getS,
+    tempPath = item.get("tempPath").getS,
+    complete = item.get("complete").getS == "true",
+    commitTime = Option(item.get("commitTime")).map(_.getN.toLong)
+  )
+}
+
+object DynamoDBLogStore {
+  val confPrefix = "spark.delta.DynamoDBLogStore."
+
+  implicit def logEntryToWrapper(entry: LogEntry): LogEntryWrapper =
+    LogEntryWrapper(entry)
+
+  def getClient(sparkConf: SparkConf): AmazonDynamoDBClient = {
+
+    val credentialsProviderName = sparkConf.get(
+      s"${confPrefix}credentials.provider",
+      "com.amazonaws.auth.DefaultAWSCredentialsProviderChain"
+    )
+    // suggested org.apache.spark.util.Utils.classForName cannot be used as is private
+    // scalastyle:off classforname
+    val authClass = Class.forName(credentialsProviderName)
+    // scalastyle:on classforname
+    val auth = authClass
+      .getConstructor()
+      .newInstance()
+      .asInstanceOf[com.amazonaws.auth.AWSCredentialsProvider];
+    var client = new AmazonDynamoDBClient(auth)
+    val regionName = sparkConf.get(s"${confPrefix}region", "us-east-1")
+    if (regionName != "") {
+      client.setRegion(Region.getRegion(Regions.fromName(regionName)))
+    }
+
+    scala.util.control.Exception.ignoring(classOf[NoSuchElementException]) {
+      client.setEndpoint(sparkConf.get(s"${confPrefix}host"))
```

Comment:
the class docs says that endpoint `defaults to 'Amazon AWS'`, can't we use that?